tidb-functions: extend tidb_decode_key docs (#5877)
pingcap/docs: functions-and-operators/tidb-functions.md

    "TIDB_DECODE_KEY(" STR ")"
```

### Examples

> In the first example we have a table `t1` which has a hidden rowid that is generated by TiDB. This is what happens for a non-clustered primary key.

Reviewer: ```suggestion
In the following example, the table `t1` has a hidden `rowid` that is generated by TiDB. The `TIDB_DECODE_KEYSQL` is used in the statement. From the result, you can see that the hidden `rowid` is decoded and output, which is a typical result for the non-clustered primary key.
```

Author: I changed `TIDB_DECODE_KEYSQL` to `TIDB_DECODE_KEY` but otherwise applied the change.

1 row in set (0.00 sec)
```

> In the second example we have a table `t2` that has a compound clustered primary key. Here the JSON output contains a `handle` that contains the name and value for both of the columns that are part of the primary key.

Reviewer: ```suggestion
In the following example, the table `t2` has a compound clustered primary key. From the JSON output, you can see a `handle` that contains the name and value for both of the columns that are part of the primary key.
```
